Scripts/modelsystem.py

```diff
@@ -271,14 +305,8 @@ def run_iteration(self, previous_iter_impedance, iteration=None):
         self.zdata_forecast["car_density"] = prediction
         self.zdata_forecast["cars_per_1000"] = 1000 * prediction
 
-        # Calculate internal demand
         self._add_internal_demand(previous_iter_impedance, iteration=="last")
-
-        # Calculate external demand
-        for mode in param.external_modes:
-            int_demand = self._sum_trips_per_zone(mode)
-            ext_demand = self.em.calc_external(mode, int_demand)
-            self.dtm.add_demand(ext_demand)
+        self._add_external_demand()
```

Comment:
At this stage, only car matrices should be added, because the other matrices are not initialized.